src/plugins/data/public/search/aggs/agg_config.ts

```diff
@@ -42,7 +46,8 @@ export type AggConfigSerialized = Ensure<
     type: string;
     enabled?: boolean;
     id?: string;
-    params?: SerializableState;
+    params?: {} | SerializableState;
+    format?: {} | Ensure<SerializedFieldFormat<SerializableState>, SerializableState>;
```

Comment:
this one is redundant, its all information is already contained in type and params

Reply:
Summarizing our offline discussion - the intent here was to return the same value that `serializeFieldFormat` currently returns, so that we could instead use this info in esaggs `formatHint` and pass it to `fieldFormats.deserialize`. I don't think this is the right long-term approach, but allows us to do this incrementally.

Next step will be looking into simplifying `fieldFormats.deserialize` so that it doesn't need to know about the special aggs-specific formats. 

And then once we agree on a design for the unified data table format, we can adjust things so that we can use the serialized aggconfig itself to look up the field format, possibly removing the need for this dedicated `format` property altogether.

Reply:
I'm removing this specifically from the serialized agg config for the time being, as it isn't strictly necessary for the purposes of this PR -- you can use `aggConfig.toSerializedFieldFormat` for now, and when we do the overhaul on the data table format we can make adjustments to the serialized agg if needed.

this means that for the time being, serialized field formats are not preserved in a serialized agg, which is fine as they were not available there previously anyway.